packages/cli/src/daemon/DaemonUtil.ts

```diff
@@ -0,0 +1,46 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import * as os from "os";
+import * as path from "path";
+import { IO } from "@zowe/imperative";
+
+/**
+ * Class containing daemon utility functions.
+ * @export
+ * @class DaemonUtil
+ */
+export class DaemonUtil {
+    /**
+     * Get the directory that holds daemon-related runtime files.
+     * Ensures that the directory exists, or we create it.
+     *
+     * @returns The absolute path to the daemon directory.
+     */
+    public static getDaemonDir(): string {
+        let daemonDir: string;
+        if (process.env?.ZOWE_DAEMON_DIR?.length > 0) {
+            // user can choose a daemon directory for storing runtime artifacts
+            daemonDir = process.env.ZOWE_DAEMON_DIR;
+        } else {
+            // our default location.
+            daemonDir = path.join(os.homedir(), ".zowe", "daemon");
```

Comment:
Is it desired behavior that we ignore the ZOWE_CLI_HOME environment variable here? My assumption is yes, since having multiple environment variables (both ZOWE_CLI_HOME and ZOWE_DAEMON_DIR) impact the daemon directory could be confusing.

Reply:
I considered using ZOWE_CLI_HOME. We have had discussions about placing plugins into a shared directory. Plugins live under ZOWE_CLI_HOME. It was unclear to me if ZOWE_CLI_HOME would be used for the new shared location, or would continue to point to user-specific data. It just seemed safer for now to ensure that the pid file (and other daemon files) be placed into a location that is specific to the user. For now, when ZOWE_CLI_HOME is left unset, it allows the daemon directory to live under the .zowe directory with all of the other Zowe stuff.

Reply:
By the way the existing files that I moved into the daemon directory were located in the user's $HOME directory, So, the new implementation is consistent with what we had before.